packages/jest-transform/src/__tests__/script_transformer.test.js

```diff
@@ -314,6 +385,68 @@ describe('ScriptTransformer', () => {
     },
   );
 
+  it(
+    "throws an error if `processAsync` doesn't return a promise of string or" +
+      'object containing `code` key with processed string',
+    () => {
+      const incorrectReturnValues = [
+        [undefined, '/fruits/banana.js'],
+        [{a: 'a'}, '/fruits/kiwi.js'],
+        [[], '/fruits/grapefruit.js'],
+      ];
+
+      const correctReturnValues = [
+        ['code', '/fruits/banana.js'],
+        [{code: 'code'}, '/fruits/kiwi.js'],
+      ];
+
+      config = {
+        ...config,
+        transform: [
+          ...incorrectReturnValues,
+          ...correctReturnValues,
+        ].map(([_, filePath]) => [
+          filePath,
+          `passthrough-preprocessor${filePath.replace(/\.|\//g, '-')}`,
+        ]),
+      };
+      const scriptTransformer = new ScriptTransformer(config);
+
+      const buildPromise = ([returnValue, filePath]) => {
+        const processorName = `passthrough-preprocessor${filePath.replace(
+          /\.|\//g,
+          '-',
+        )}`;
+
+        jest.doMock(
+          processorName,
+          () => ({
+            processAsync: jest.fn(),
+          }),
+          {virtual: true},
+        );
+
+        const transformer = require(processorName);
+        transformer.processAsync.mockResolvedValue(returnValue);
+
+        return scriptTransformer.transformAsync(filePath, {});
+      };
+
+      expect.assertions(
+        incorrectReturnValues.length + correctReturnValues.length,
+      );
+
+      incorrectReturnValues.map(buildPromise).forEach(async promise => {
+        // Jest must throw error
+        await expect(promise).rejects.toThrow();
+      });
+
+      correctReturnValues.map(buildPromise).forEach(async promise => {
+        await expect(promise).resolves.toHaveProperty('code');
+      });
```

Comment:
```suggestion
      await Promise.all([
        incorrectReturnValues.map(buildPromise).map(async promise => {
          // Jest must throw error
          await expect(promise).rejects.toThrow();
        }),
      ]);
      await Promise.all([
        correctReturnValues.map(buildPromise).map(async promise => {
          await expect(promise).resolves.toHaveProperty("code");
        }),
      ]);
```

you need to `await` in the outer scope as well, otherwise the test will not wait for the assertions

Reply:
Even when test does not wait for the assertions, `expect.assertions()` on line 436 still recognise these assertions?

Reply:
sure, but they might actually pass and we just didn't wait long enough, makes it sorta flakey. We don't need the `expect.assertions` with the `await` now, but it doesn't hurt either

Reply:
Will `Promise.all()` implicitly trigger additional assertions? 

When [only asserting 3 incorrect promises would be rejected ](https://github.com/ychi/jest/blob/e70948780eb3a0eef3a2455f3335a9757c0be9ca/packages/jest-transform/src/__tests__/script_transformer.test.ts#L449-L461), there were [6 assertions](https://github.com/ychi/jest/runs/1380812645?check_suite_focus=true#step:6:35); while asserting [2 correct promises](https://github.com/ychi/jest/blob/53420539392f38fda780cd2fbff55a30392ecc65/packages/jest-transform/src/__tests__/script_transformer.test.ts#L458-L462) to be resolved, the assertion count would be [4](https://github.com/ychi/jest/runs/1380885257?check_suite_focus=true#step:6:35)